src/libstd/macros.rs

```diff
@@ -7,20 +7,19 @@
 #[doc(include = "../libcore/macros/panic.md")]
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
-#[allow_internal_unstable(libstd_sys_internals)]
+#[allow_internal_unstable(libstd_sys_internals, track_caller)]
 macro_rules! panic {
     () => ({
         $crate::panic!("explicit panic")
     });
     ($msg:expr) => ({
-        $crate::rt::begin_panic($msg, &($crate::file!(), $crate::line!(), $crate::column!()))
+        $crate::rt::begin_panic($msg, #[cfg(bootstrap)] &(file!(), line!(), column!()))
```

Comment:
You can't use `#[cfg]` inside macros, because it will be checked when the macro is used, you *really* want to have two definitions of the macro (sorry!).

Reply:
Why not? This works just fine on CI and I'd expect it so since it's just expanding source code

Reply:
It's not, though, we'd be shipping `libstd`s with `#[cfg(bootstrap)]` in them!
`bootstrap` is scoped to our build system, but if it's inside the macro expansion like this, suddenly users can affect it with `--cfg bootstrap` (which they can't with anything else).

Reply:
K -- I'm almost done fixing this. It'd be good to put that in the guide somewhere, navigating `#[cfg(bootstrap)]` has been interesting to say the least :P

Reply:
https://github.com/rust-lang/rustc-guide/issues/552